swap/src/bin/swap.rs

```diff
@@ -39,7 +39,24 @@ extern crate prettytable;
 
 #[tokio::main]
 async fn main() -> Result<()> {
-    let Arguments { data, debug, cmd } = Arguments::from_args();
+    let Arguments {
+        testnet,
+        data,
+        debug,
+        cmd,
+    } = Arguments::from_args();
+
+    let (data_dir, env_config) = if testnet {
+        let data_dir = data.0.join("testnet");
+        let env_config = env::Testnet::get_config();
+
+        (data_dir, env_config)
+    } else {
+        let data_dir = data.0.join("mainnet");
+        let env_config = env::Mainnet::get_config();
+
+        (data_dir, env_config)
+    };
```

Comment:
Nice 👍

Reply:
Ideally this functionality doesn't live inside `main.rs` though because it is hard to test an noisy.